lambdaclass/mirra_backend
# [GH-601] Track more match data

## Motivation

Closes #601 

## Summary of changes

- [x] Rebase after #578 is merged
- [x] Retry mechanism when reporting result
- More things to track
  - [x] Damage done
  - [x] Damage taken
  - [x] Health healed
  - [x] What character killed you
  - [x] Killed by human or bot?
  - [x] Match duration

## How to test it?

To test you can run a game through the browser (`game_client`) by going to [localhost:3000](http://localhost:3000/) and login or if you don't have the Google credentials you can create a user `GameBackend.Users.find_or_create_google_user_by_email("tester")` and then go to `http://localhost:3000/board/<YOUR_USER_ID>/muflus/player_name/quick_game`

Then after the match ends you can go to the console and see the results by doing

```
GameBackend.Repo.all(GameBackend.Matches.ArenaMatchResult)
```


## Checklist
